integration-test-groups/aws2/aws2-cw/src/test/java/org/apache/camel/quarkus/component/aws2/cw/it/Aws2CwTest.java

```diff
@@ -90,4 +97,148 @@ public void metric() {
+                                    .dimensions(Dimension.builder().name("type").value("even").build())
+                                    .endTime(Instant.ofEpochMilli(System.currentTimeMillis() + 10000))
+                                    .period(30)
+                                    .build())
+                            .datapoints();
+                    LOG.debug("Expecting some datapoints for metric " + namespace + "/" + metricName + " (type='odd'), got "
+                            + oddDatapoints);
+                    if (oddDatapoints.isEmpty()) {
+                        return false;
+                    }
+                    LOG.debug("Expecting some datapoints for metric " + namespace + "/" + metricName + " (type='even'), got "
+                            + evenDatapoints);
+                    if (evenDatapoints.isEmpty()) {
+                        return false;
+                    }
+
+                    Datapoint oddDp = oddDatapoints.get(0);
+                    Datapoint evenDp = evenDatapoints.get(0);
+
+                    if (!(oddDp.sampleCount().intValue() == 2
+                            && oddDp.minimum().intValue() == 2 * value + 1
+                            && oddDp.maximum().intValue() == 2 * value + 3)) {
+                        throw new RuntimeException("Unexpected odd datapoint " + oddDp
+                                + "; expected sampleCount == 2 && minimum ~ " + (2 * value + 1)
+                                + " && maximum ~ " + (2 * value + 3));
+                    }
+
+                    if (!(evenDp.average() == 2 * value + 2
+                            && evenDp.sampleCount() == 3)) {
+                        throw new RuntimeException("Unexpected even datapoint " + evenDp
+                                + "; expected sampleCount == 3 && average == " + (2 * value + 2));
+                    }
+
+                    return true;
+                });
+
+    }
+
+    @Test
+    public void customClient() {
+        final String namespace = "cq-metrics-" + java.util.UUID.randomUUID().toString().replace("-", "");
+        final String metricName = "metricName" + java.util.UUID.randomUUID().toString().replace("-", "");
+        final int value = (int) (Math.random() * 10000);
+
+        List<Map<String, Object>> data = new LinkedList<>();
+
+        data.add(CollectionHelper.mapOf(
+                Cw2Constants.METRIC_NAMESPACE, namespace,
+                Cw2Constants.METRIC_NAME, metricName,
+                Cw2Constants.METRIC_VALUE, 2 * value,
+                Cw2Constants.METRIC_UNIT, "Count"));
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .queryParam("customClientName", "customClient")
+                .body(data)
+                .post("/aws2-cw/send-metric-maps/" + namespace)
+                .then()
+                .statusCode(200)
+                .body(is("MockedClient"));
```

Comment:
@aldettinger Here is assertion for the result of the mocked client exception from previous comment.

Reply:
ok, caught. Many thanks.